Cleaned Experiment/views.py

meshde opened the pull request:
Cleared all references to Board,Slot and Booking models in experiment/views.py. Also changed the Experiment model schema to include Foreign Key reference to Account.(Previously, Experiment had foreign key to Booking and Booking in turn had foreign key to Account)

The client is successfully able to connect to the modified local django server on my PC with the message displayed as "Ready to execute Scilab code"

meshde committed:
Commented sbhs_server.admin in settings

meshde committed:
Further Testing and Commenting import statements of yaksh

meshde committed:
Delete Board and all it's references.Remake Database.Change initiation code

meshde committed:
Remove Bookings Foreign Key in Experiment model

meshde committed:
Removed all references to Slots and Bookings in exxperiment.views. Added a Foreign Key Reference to Account inthe Experiment Model

meshde committed:
Cleaned experiments/views.py code without referneces to Board or Booking or Slot models

meshde committed:
Change SBHS to output Sum of Inputs

meshde committed:
Minor Fixes

meshde commented:
I have also completely changed the sbhs.py file, the getTemp() method now returns the sum of the fan and heat percentages.
The code is working with the scilab experiments and correct output is being displayed in the client application window

I have just commented out the previous codes rather than deleting them.This is to facilitate quick debugging and fixing, I'll be sure to completely clean the code once all possible changes have been made and the code works perfectly

maheshgudi commented on experiments/meshde/2017Feb24_17_32_51.txt:
```diff
@@ -0,0 +1,14 @@
+3 30 50 80.00 1487937848676 1487937848684 1487937848720 0.30000E+01
+4 30 50 80.00 1487937848847 1487937848859 1487937848896 0.40000E+01
+5 30 50 80.00 1487937849021 1487937849028 1487937849064 0.50000E+01
+6 30 50 80.00 1487937849189 1487937849199 1487937849239 0.60000E+01
```
Please get rid of these files. They are not useful to a developer.

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -3,7 +3,7 @@
 from django.contrib.auth import login as LOGIN
-from sbhs_server.tables.models import Slot, Account, Experiment, Booking
+from sbhs_server.tables.models import Account, Experiment
 import json, datetime, os, time
```
Import the modules individually.

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -18,61 +18,33 @@ def initiation(req):
+            e.log=os.path.join(logdir, filename)
+            e.save()
+
+            #key = str(1)#Temporarily till SBHS class and settings.board dictionary are changed
```
Get rid of the commented code

maheshgudi commented on sbhs_server/tables/migrations/0008_initial.py:
```diff
@@ -8,17 +8,6 @@
 class Migration(SchemaMigration):
 
     def forwards(self, orm):
```
Get rid of the migration files. They are not needed
 Add them in gitignore

meshde committed:
Cleaned SBHS code

meshde committed:
Remove Webcam and Slot folders

meshde committed:
Remove all references to Bookig,Slots and Boards from all python files

meshde committed:
Change account/sub_nav template to exclude Bookings and Slot viewing

meshde committed:
Changed Account Schema to include Sbhs object for each user

meshde committed:
Cleaned Code and removed SBHS object attribute from Account

meshde commented:
I've removed each and every reference to Slot,Webcam,Bookings, and board.
The site and the client experiment are working perfectly
I've also added the suggested directories in the gitignore, but they still have been pushed. Could you please check if I did anything wrong with respect to the gitignore file?

maheshgudi commented on .gitignore:
```diff
@@ -0,0 +1,5 @@
+experiments\
+log\
+*.sqlite3
+migrations\
```
add a new line here. All files should end on a new line.

meshde replied:
You mean a blank last line?

maheshgudi replied:
yes

maheshgudi commented on sbhs_server/admin/views.py:
```diff
@@ -1,7 +1,7 @@
 from django.http import Http404
 from django.contrib.auth.decorators import login_required
-from sbhs_server.tables.models import Board, Booking
+#from sbhs_server.tables.models import Board, Booking
```
commented code.

maheshgudi commented on sbhs_server/admin/views.py:
```diff
@@ -10,30 +10,30 @@ def checkadmin(req):
-    boards = Board.objects.order_by('online').all()
-    allotment_mode = "Random" if Board.can_do_random_allotment() else "Workshop"
-    return render(req, 'admin/index.html', {"boards": boards, "allotment_mode": allotment_mode})
+#@login_required(redirect_field_name=None)
```
Commented code.

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -3,76 +3,47 @@
 from django.contrib.auth import login as LOGIN
-from sbhs_server.tables.models import Slot, Account, Experiment, Booking
+from sbhs_server.tables.models import Account, Experiment
 import json, datetime, os, time
```
Please import the modules separately as asked.

meshde replied:
I'm sorry, forgot about these.I'll change them right away

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -3,76 +3,47 @@
 # Create your views here.
 # 
+
+global boards
```
What is boards? Why are you putting that in global namespace?

meshde replied:
'boards' is a Sbhs object that is created when the user logs in from the client application in the initiation() method. I've declared it global as the object created during initiation() is being used in the experiment() method.

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -3,76 +3,47 @@
 
 @csrf_exempt
 def initiation(req):
+    global boards
```
Again? Why this?

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -87,77 +58,67 @@ def initiation(req):
-            STATUS = 0
-            MESSAGE = "You haven't booked this slot."
-
+        global boards
```
Why?

maheshgudi commented on sbhs_server/experiment/views.py:
```diff
@@ -87,77 +58,67 @@ def initiation(req):
+
+            experiment = experiment[0]
+            now = datetime.datetime.now()
+            #boards.setHeat(0)
```
commented code

maheshgudi commented on sbhs_server/settings.py:
```diff
@@ -203,8 +201,8 @@
 EXPERIMENT_LOGS_DIR = os.path.join(BASE_DIR, 'experiments')
-WEBCAM_DIR = os.path.join(STATIC_ROOT, 'img/webcam/') if is_production else os.path.join(BASE_DIR, 'static/img/webcam/')
-WEBCAM_STATIC_DIR = 'img/webcam/'
+#WEBCAM_DIR = os.path.join(STATIC_ROOT, 'img/webcam/') if is_production else os.path.join(BASE_DIR, 'static/img/webcam/')
```
get rid of the code.

maheshgudi reviewed (changes requested):
Need to clean up the code further.

meshde committed:
Cleared commented codes and separated import statements

meshde committed:
Removed Global boards

meshde committed:
Changed gitignore

meshde committed:
Minor fix in experiments.views.log_data

meshde committed:
Changed Sbhs.py file to reference formula.py file to calculate temperature. formula.py contains dummy formula

meshde committed:
Changed timeleft to inf

meshde committed:
Changed timeleft to 99 minutes 59 seconds

meshde committed:
Fixed error that prevented log files from being downloaded

meshde committed:
Minor Changes in sbhs_formula

meshde committed:
Added id of the coefficient set used for the experiment as coeff_ID

meshde committed:
Added getID() method to return coefficient set ID

meshde committed:
log_data now takes Coefficient Set ID instead of Machine ID

meshde commented:
I've made all changes that I was asked to make. The things that remain to be done include changing the admin views to display some relevant info(I haven't been told what to do about it) and integrating with yaksh module.

meshde committed:
Changed Database Structure. coeff_ID is now assigned to each user during account creation

rupakrokade commented:
Merged mehmoods work with the base repo
